plugins/org.eclipse.fordiac.ide.model.edit/src/org/eclipse/fordiac/ide/model/edit/TypeEntryAdapter.java

```diff
@@ -146,18 +147,16 @@ private void handleDependencyUpdate(final TypeEntry typeEntry) {
 
 	private static void handleAttributeTypeEntryUpdate(final LibraryElement editedElement,
 			final AttributeTypeEntry atEntry) {
-		final AttributeTypeInstanceSearch search = new AttributeTypeInstanceSearch(atEntry);
+		final AttributeTypeInstanceSearch search = new AttributeTypeInstanceSearch(editedElement, atEntry);
 		final List<? extends EObject> result = search.performSearch();
 
 		result.forEach(at -> {
 			// update attribute here
 			if (at instanceof final ConfigurableObject co) {
-				final UpdateAttributeDeclarationCommand cmd = new UpdateAttributeDeclarationCommand(co, atEntry);
-				cmd.execute();
+				final Attribute attribute = co.getAttribute(atEntry.getTypeName());
+				ChangeAttributeDeclarationCommand.forName(attribute, atEntry.getTypeName()).execute();
```

Comment:
This may break depending on how the type name is used, that is, with or without package name. I would suggest extracting a `forTypeEntry(AttributeTypeEntry)` method from: https://github.com/eclipse-4diac/4diac-ide/blob/9aa7a522ec56eebea9c4629d9857d6e0726ef8e4/plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/ChangeAttributeDeclarationCommand.java#L67-L73

Reply:
Good point. Done.